src/ZkFold/Base/Algebra/Basic/Class.hs

```diff
@@ -279,9 +279,10 @@ If @a@ and @b@ are in @R@ and @b@ is nonzero, then there exist @q@ and @r@ in @R
 The function @divMod@ associated with this class produces @q@ and @r@ given @a@ and @b@.
 -}
 class Semiring a => EuclideanDomain a where
```

Comment:
Let's rename this to `SemiEuclidean` in this PR? `Euclidean(Domain)` should be `Ring` + `SemiEuclidean` + `eea`